File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> numbers, int target) {
    int answer = 0;
    queue<int>q;
    q.push(0);
    for(int idx = 0; idx < numbers.size(); idx++){
        queue<int>tmpQueue;
        while(!q.empty()){
            int tmpSum = q.front();
            tmpQueue.push(tmpSum + numbers[idx]);
            tmpQueue.push(tmpSum - numbers[idx]);
            q.pop();    
        }
        q = tmpQueue;   
    }
    
    while(!q.empty()){
        if(q.front() == target) answer++;
        q.pop();
    }
    
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[] numbers, int target) {
        Queue<Integer> q = new LinkedList<>();
        q.add(0);
        for(int num : numbers){
            int size = q.size();
            for(int i = 0; i < size; i++){
                int head = q.poll();
                q.add(head+num);
                q.add(head-num);
            }
        }
        return (int)q.stream().filter(i -> i == target).count();
    }
}